File: Funcional/src/Menu.hs
module Menu
  ( run,
    menu,
  )
where

import Album
import Loja
import Bafo
import System.Exit
import Text.Printf
import Util


run :: IO()
run = do
  logo
  mensagemInicial


logo :: IO ()
logo = do
  clearScr
  putStrLn "        =============================================      "
  putStrLn "              ██████╗ ██████╗ ███████╗██╗ ██████╗          "
  putStrLn "             ██╔════╝██╔═══██╗██╔════╝██║██╔════╝          "
  putStrLn "             ██║     ██║   ██║█████╗  ██║██║  ███╗         "
  putStrLn "             ██║     ██║   ██║██╔══╝  ██║██║   ██║         "
  putStrLn "             ╚██████╗╚██████╔╝██║     ██║╚██████╔╝         "
  putStrLn "              ╚═════╝ ╚═════╝ ╚═╝     ╚═╝ ╚═════╝          "
  putStrLn "        =============================================      "

mensagemInicial :: IO ()
mensagemInicial = do
  putStrLn "                                                                "
  putStrLn "  <<Seja bem vindo ao seu simulador de Álbum da Copa do Mundo>> "
  putStrLn "                       <<Deseja iniciar?>>                      "
  putStrLn "                              1.SIM                             "
  putStrLn "                              2.NÃO                             "
  putStrLn "                                                                "
  entradaIniciaJogo

menu :: IO ()
menu = do
  clearScr
  putStrLn "        ================================================      "
  putStrLn "             ███╗   ███╗███████╗███╗   ██╗██╗   ██╗           "
  putStrLn "             ████╗ ████║██╔════╝████╗  ██║██║   ██║           "
  putStrLn "             ██╔████╔██║█████╗  ██╔██╗ ██║██║   ██║           "
  putStrLn "             ██║╚██╔╝██║██╔══╝  ██║╚██╗██║██║   ██║           "
  putStrLn "             ██║ ╚═╝ ██║███████╗██║ ╚████║╚██████╔╝           "
  putStrLn "             ╚═╝     ╚═╝╚══════╝╚═╝  ╚═══╝ ╚═════╝            "
  putStrLn "        ================================================      "
  menuPrincipal

menuPrincipal :: IO ()
menuPrincipal = do
  conteudo <- lerArquivo "/src/arquivos/album.csv"
  estaCompleto conteudo
  putStrLn "                                               "
  putStrLn "                             1. ALBUM          "
  putStrLn "                             2. LOJA           "
  putStrLn "                             3. BAFO           "
  putStrLn "                             4. FINALIZAR      "
  putStrLn "                                               "
  entradaMenuPrincipal


entradaMenuPrincipal :: IO()
entradaMenuPrincipal = do
  opc <- getLine :: IO String
  navegacaoMenu opc

navegacaoMenu :: String -> IO()
navegacaoMenu opc
  | opc == "1" = do 
    album
    menu
  | opc == "2" = do 
    loja
    menu
  | opc == "3" = do 
    bafo
    menu
  | opc == "4" = exitSuccess
  | otherwise = do
    putStrLn "\nDigite uma opção válida\n"
    entradaMenuPrincipal


entradaIniciaJogo :: IO()
entradaIniciaJogo = do
  opc <- getLine :: IO String
  navegacaoIniciar opc


navegacaoIniciar :: String -> IO()
navegacaoIniciar opc
  | opc == "1" = menu
  | opc == "2" = exitSuccess
  | otherwise = do
    putStrLn "\nDigite uma opção válida\n"
    entradaIniciaJogo
          
estaCompleto:: [String] -> IO()
estaCompleto [] = do 
  putStr "\nParabéns você completou o álbum\nObrigado por jogar\n"
  exitSuccess
estaCompleto (a:as) = if a == "True" then estaCompleto as
  else putStrLn ""

File: Logico/menu.pl
:- include('loja.pl').
:- include('album.pl').
:- include('bafo.pl').
:- include('util.pl').

finalizaJogo :-
getAlbum(Album),
contaFigurinhas(Album, Possui, NPossui),
(NPossui == 0 ->
writeln("\nParabéns, você conseguiu todas as figurinhas!"),continuar,halt;
writeln("")
).

menu :-

    finalizaJogo,shell(clear),
    writeln("                                                              "),
    writeln("        ================================================      "),
    writeln("             ███╗   ███╗███████╗███╗   ██╗██╗   ██╗           "),
    writeln("             ████╗ ████║██╔════╝████╗  ██║██║   ██║           "),
    writeln("             ██╔████╔██║█████╗  ██╔██╗ ██║██║   ██║           "),
    writeln("             ██║╚██╔╝██║██╔══╝  ██║╚██╗██║██║   ██║           "),
    writeln("             ██║ ╚═╝ ██║███████╗██║ ╚████║╚██████╔╝           "),
    writeln("             ╚═╝     ╚═╝╚══════╝╚═╝  ╚═══╝ ╚═════╝            "),
    writeln("        ================================================      "),
    menuPrincipal.



menuPrincipal :-
    getDinheiro(Dinheiro), getRepetidas(Repetidas), write("Dinheiro: "), write(Dinheiro), write("  Repetidas: "), writeln(Repetidas),
    writeln("                                               "),
    writeln("                             1. ALBUM          "),
    writeln("                             2. LOJA           "),
    writeln("                             3. BAFO           "),
    writeln("                             4. FINALIZAR      "),
    writeln("                                               "),
    
    entradaMenuPrincipal.



entradaMenuPrincipal:- 
    read_line_to_string(user_input, Opc),
    navegacaoMenu(Opc).

navegacaoMenu("1") :- album, menu.
navegacaoMenu("2") :- loja, menu.
navegacaoMenu("3") :- bafo, menu.
navegacaoMenu("4") :- halt.

navegacaoMenu(_) :- write("Digite uma opção válida\n"), entradaMenuPrincipal.
